utils.py

```diff
@@ -5,33 +5,142 @@
+                detect_num = int(i_line.rsplit(' ')[0])
+            elif 'External MUX Channels' in i_line:
+                mux_num = int(i_line.rsplit(' ')[0])
+            elif 'Auxiliary Channels' in i_line:
+                aux_num = int(i_line.rsplit(' ')[0])
+            elif 'Waveform (CCF) Frequency (Hz)' in i_line:
+                ccf_ha = float(i_line.rsplit(' ')[0])
+            elif 'Update Rate (Hz)' in i_line:
+                srate = float(i_line.rsplit(' ')[0])
+            elif 'Updata Rate (Hz)' in i_line:
+                srate = float(i_line.rsplit(' ')[0])
+            elif '#DATA BEGINS' in i_line:
+                start_line = line_num
 
-import mne
-
-def boxy2mne(mtg_file,tol_file):
+    ###now let's go through and parse our raw data###  
+    raw_data = pd.read_csv(boxy_file, skiprows=start_line, sep='\t')
     
-    ###need to have function determine sampling rate automatically, eventually
-    srate = 39.0625
+    ###only go up to 'J' since that the max number of detectors we can have###
+    ###for now anyway###
+    detectors = ['A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 
+                 'M', 'N', 'O', 'P', 'Q', 'R', 'S', 'T', 'U', 'V', 'W', 'X',
+                 'Y', 'Z']
+    data_types = ['AC','DC','Ph']
+    sources = np.arange(1,mux_num+1,1)
+    
+    ###since we can save boxy files in two different styles###
+    ###this will check to see which style the data is saved###
+    ###seems to also work with older boxy files###
+    if 'exmux' in raw_data.columns:
+        filetype = 1
+        
+        ###drop the last line as this is just '#DATA ENDS'###
+        raw_data = raw_data.drop([len(raw_data)-1])
+        
+        ###store some extra info###
+        record = raw_data['record'].to_numpy()
+        exmux = raw_data['exmux'].to_numpy()
+        
+        ###make some empty variables to store our data###
+        raw_ac = np.zeros((detect_num*mux_num,int(len(raw_data)/mux_num)))
+        raw_dc = np.zeros((detect_num*mux_num,int(len(raw_data)/mux_num)))
+        raw_ph = np.zeros((detect_num*mux_num,int(len(raw_data)/mux_num)))
+    else:
+        filetype = 2
+        
+        ###drop the last line as this is just '#DATA ENDS'###
+        ###also drop the first line since this is empty###
+        raw_data = raw_data.drop([0,len(raw_data)-1])
+        
+        ###make some empty variables to store our data###
+        raw_ac = np.zeros(((detect_num*mux_num),len(raw_data)))
+        raw_dc = np.zeros(((detect_num*mux_num),len(raw_data)))
+        raw_ph = np.zeros(((detect_num*mux_num),len(raw_data)))
+    
+    ###store some extra data, might not need these though###
+    time = raw_data['time'].to_numpy()
+    group = raw_data['group'].to_numpy()
+    step = raw_data['step'].to_numpy()
+    mark = raw_data['mark'].to_numpy()
+    flag = raw_data['flag'].to_numpy()
+    aux1 = raw_data['aux-1'].to_numpy()
+    digaux = raw_data['digaux'].to_numpy()
```

Comment:
this will be where the markers are, so we want to load those in with the data and save in the Raw format correct location

Reply:
okay, I will add that